lib/psych/visitors/yaml_tree.rb

```diff
@@ -17,19 +17,15 @@ class Registrar # :nodoc:
         def initialize
           @obj_to_id   = {}.compare_by_identity
           @obj_to_node = {}.compare_by_identity
-          @targets     = []
           @counter     = 0
         end
 
         def register target, node
-          @targets << target
           @obj_to_node[target] = node
         end
 
         def key? target
           @obj_to_node.key? target
-        rescue NoMethodError
-          false
```

Comment:
I missed this in #663.

I believe it existed to just catch if `object_id` was not defined, but since we don't call that anymore, we don't have to worry about it being missing.